### DISCUSSION TITLE
Very chatty when trying to reconnect

### DISCUSSION THREAD

**Discussion by cherio** [erebe/wstunnel#266] [General]
When the server goes down, the client frequently attempts to re-connect.
Is there a way to control the frequency or the reconnect behavior in general?

Ideally (I understand this may be specific to my environment and needs) I'd like the client to try to reconnect frequently at first but then gradually reduce the frequency of connection attempts to a certain specified period (In my case it would once per minute).

**Comment by erebe** [comment]
Hello,

You qre in luck, this is the next feature that is planned ;P
I should add an exponential backoff with a config to be to able to clamp the max value.

**Reply by cherio** [reply]
That's it!!

**Reply by erebe** [reply]
it is available in the latest release https://github.com/erebe/wstunnel/releases/tag/v9.4.2
with --connection-retry-max-backoff-sec , but the default should be good enough
